pkg/supervisor/supervisor.go

```diff
@@ -209,6 +209,8 @@ func (su *Supervisor) waitStatesTimeout(to time.Duration) (func() error, error)
 			}
 		}()
 
+		// With non-zero timeout parameter, call should be aware that receiver is nil
```

Comment:
Can we explain why the nil receiver will be returned if the timeout is zero?

Reply:
fixed